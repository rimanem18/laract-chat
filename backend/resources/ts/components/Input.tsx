import { TextField } from '@mui/material'
import React from 'react'

type InputProps = {
  label: string
  type: string
  name: string
  value: string
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}
const Input = ({ label, type, name, value, onChange }: InputProps) => {
  return (
    <>
      <div className="form-group">
        <TextField
          id={label}
          label={label}
          variant="standard"
          fullWidth
          margin="normal"
          type={type}
          name={name}
          value={value}
          onChange={onChange}
          data-testid={`${label}-input`}
        />
      </div>
    </>
  )
}

export default React.memo(Input)
