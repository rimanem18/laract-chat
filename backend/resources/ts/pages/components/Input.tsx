import React from 'react'

type InputProps = {
  label: string
  type: string
  name: string
  value: string
  onChange: React.ChangeEventHandler<HTMLInputElement>
}
const Input = ({ label, type, name, value, onChange }: InputProps) => {
  return (
    <>
      <div className="form-group">
        <label htmlFor={label} className="form-label">
          {label}
        </label>
        <input
          id={label}
          className="form-control"
          type={type}
          name={name}
          value={value}
          onChange={onChange}
        />
      </div>
    </>
  )
}

export default React.memo(Input)
