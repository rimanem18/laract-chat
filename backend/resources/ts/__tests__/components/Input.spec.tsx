import React from 'react'
import '@testing-library/jest-dom'
import { fireEvent, render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Input from '../../components/Input'

// Hooks の Mock
const mockUseAppDispatch = jest.fn()
const mockUseAppSelector = jest.fn()
jest.mock('../../app/hooks', () => ({
  useAppDispatch:
    () =>
    (...args: any[]) =>
      mockUseAppDispatch(...args),
  useAppSelector:
    () =>
    (...args: any[]) =>
      mockUseAppSelector(...args),
}))

// Mock の定義
const onChangeMock = jest.fn()

const InputComponent = (
  <Input
    label="Name"
    type="text"
    name="name"
    value=""
    onChange={onChangeMock}
  />
)

// Setup
const setup = () => {
  const screen = render(InputComponent)
  const input = screen.getByTestId('Name-input') as HTMLInputElement
  const label = screen.getByTestId('label') as HTMLLabelElement
  return {
    input,
    label,
    ...screen,
  }
}

// テスト開始
describe('Input', () => {
  it('snapshot', () => {
    const tree = renderer.create(InputComponent).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('渡されたラベルの値が htmlFor, text に反映されている', () => {
    const { label } = setup()

    expect(label.htmlFor).toBe('Name')
    expect(label.textContent).toBe('Name')
  })

  it('渡された type の値が反映されている', () => {
    const { input } = setup()

    expect(input.type).toBe('text')
  })

  it('渡された name の値が name に反映されている', () => {
    const { input } = setup()

    expect(input.name).toBe('name')
  })

  it('渡された value の文字が反映されている', () => {
    const { input } = setup()

    expect(input.value).toBe('')
  })

  it('value が変化すると onChange が発火する', () => {
    const { input } = setup()

    fireEvent.change(input, {
      target: { value: 'Hello' },
    })
    expect(onChangeMock).toBeCalled()
  })
})
