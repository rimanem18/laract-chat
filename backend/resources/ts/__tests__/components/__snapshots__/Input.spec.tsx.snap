// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot 1`] = `
<div
  className="form-group"
>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="Name"
      id="Name-label"
    >
      Name
    </label>
    <div
      className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-ghsjzk-MuiInputBase-root-MuiInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        data-testid="Name-input"
        disabled={false}
        id="Name"
        name="name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
    
  </div>
</div>
`;
